src/rdb.c

```diff
@@ -3351,6 +3358,12 @@ int rdbLoad(char *filename, rdbSaveInfo *rsi, int rdbflags) {
 
     fclose(fp);
     stopLoading(retval==C_OK);
+    /* Reclaim the cache backed by rdb */
+    if (retval == C_OK && !(rdbflags & RDBFLAGS_KEEP_CACHE)) {
+        /* TODO: maybe we could combine the fopen and open into one in the future */
+        rdb_fd = open(server.rdb_filename, O_RDONLY);
+        if (rdb_fd > 0) bioCreateCloseJob(rdb_fd, 0, 1);
```

Comment:
fsync should coordinate with write to guarantee the safety of file, rdbLoad just read the file, I think it's based on a assumption that the file has been fsynced at the time write is performed.

Reply:
ok, indeed both rdbSave and readSyncBulkPayload do an fsync before leaving, so we're safe there.
but on the other hand, it means that adding another fsync would be harmless, so it may be "safer" or cleaner to always make sure we couple an fadvise with an fsync just before it.
we can do that by passing `1` as the need_fsync, or by modifying bio.c to always force an fsync before calling fadvise, even if need_sync is 0.
what do you think?

Reply:
I still don't think it's a good idea. fsync belongs to the scope of write, but we do something related to write while reading a file, the couple of read and write looks a little odd.
Another problem is, while fadvise has no side effect to write, but fsync do. While a page is marked as writeback, the page can not be written util the writeback ends(more precisely the write would block util writeback is done), so fsync here may interfere a write at other place, which is unexpected.

Reply:
ok, let's leave it as is then.